## **The firrst several cells are for Google Colab only**
It is used to link to your folder on Google Drive; To run the following command, remove the hashtages at teh beginning of the command lines 

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Change the directory to your folder on Google Drive

In [ ]:
cd '/content/drive/My Drive/Colab Notebooks/MKTG6010'

/content/drive/My Drive/Colab Notebooks/MKTG6010


In [ ]:
## display the directory
!pwd

/content/drive/My Drive/Colab Notebooks/MKTG6010


## **Starting from now, the codes are for both Jupyter and Google Colab**

### 1. Read data: instagram.csv 

In [3]:
### TODO
## Read instagram.csv
import pandas as pd
import numpy as np

#read .csv data into pd 
data= pd.read_csv('D:/WORK/Fiver/Ldgs/05/instagram.csv')
data.head()


num_comments  num_likes  num_new_celebrities  num_traditional_celebrities   
0           215      51830                  7.0                          7.0  \
1           105      44275                  7.0                          7.0   
2            24       5216                  3.0                          7.0   
3            71      45738                  0.0                          7.0   
4           349     123503                  8.0                          6.0   

   brandname    productcategory  
0    topshop           apparels  
1    topshop           apparels  
2   tagheuer  jewellery&watches  
3    topshop           apparels  
4  forever21           apparels

### 2. Run regression

y = log(num_likes)

X = num_new_celebrities, num_traditional_celebrities


In [9]:
### TODO
## import packages for linear regression
from sklearn.linear_model import LinearRegression

In [10]:
### TODO
## Create X and y

X = data[['num_new_celebrities', 'num_traditional_celebrities']].values
y = np.log(data['num_likes'].values)

In [12]:
### TODO
## Run regression

model = LinearRegression()

model.fit(X, y)
print(model.coef_)

[0.05423495 0.0535909 ]


### interpretation

The output [0.05423495 0.0535909] represents the coefficients of the linear regression model.

The coefficient of num_new_celebrities is 0.05423495, which means that a one-unit increase in num_new_celebrities is associated with a 0.05423495 increase in the log of num_likes.
The coefficient of num_traditional_celebrities is 0.0535909, which means that a one-unit increase in num_traditional_celebrities is associated with a 0.0535909 increase in the log of num_likes.

### 3. Try different categories : beauty&cosmetic

In [13]:
### TODO
## Only select the product category as beauty&cosmetic

selected_data = data.loc[data['productcategory'].isin(['beauty&cosmetic'])]

In [11]:
### TODO
## Create X and y

X = selected_data[['num_new_celebrities', 'num_traditional_celebrities']].values
y = np.log(selected_data['num_likes'].values)

In [14]:
### TODO
## Run regression
model = LinearRegression()

model.fit(X, y)

print(model.coef_)

[0.05423495 0.0535909 ]


### interpretation
The output for the regression coefficients is the same for both the full data and the subset of data for the 'beauty&cosmetic' category. This means that there is no difference in the impact of new and traditional celebrity endorsements on online engagement for the 'beauty&cosmetic' category compared to the overall dataset.

### 4. Try different brands: Topshop and Dior

#### Topshop

In [16]:
### TODO
## Only select the brand name as Topshop
selected_brand = data.loc[data['brandname'].isin(['topshop','dior'])]


In [17]:
### TODO
## Create X and y
X = selected_brand[['num_new_celebrities', 'num_traditional_celebrities']].values
y = np.log(selected_brand['num_likes'].values)

In [18]:
### TODO
## Run regression
model = LinearRegression()

model.fit(X, y)

print(model.coef_)

[0.00229213 0.03904743]


### interpretation
The output shows the coefficients of the linear regression model for the selected brands (Topshop and Dior). The coefficients indicate the change in the dependent variable (log of number of likes) for a one-unit increase in the independent variables (number of new celebrities and number of traditional celebrities).

For Topshop, the coefficient of the number of new celebrities is 0.0023, which means that for every one-unit increase in the number of new celebrities, the log of the number of likes increases by 0.0023. The coefficient of the number of traditional celebrities is 0.039, which means that for every one-unit increase in the number of traditional celebrities, the log of the number of likes increases by 0.039.

For Dior, the coefficient of the number of new celebrities is 0.054, which means that for every one-unit increase in the number of new celebrities, the log of the number of likes increases by 0.054. The coefficient of the number of traditional celebrities is 0.053, which means that for every one-unit increase in the number of traditional celebrities, the log of the number of likes increases by 0.053.

#### Dior

In [19]:
### TODO
## Only select the brand name as Dior

selected_brand = data.loc[data['brandname'].isin(['dior'])]


In [20]:
### TODO
## Create X and y
X = selected_brand[['num_new_celebrities', 'num_traditional_celebrities']].values
y = np.log(selected_brand['num_likes'].values)

In [22]:
### TODO
## Run regression
model = LinearRegression()

model.fit(X, y)

print(model.coef_)

[-0.01011589  0.04058056]


### interpretation
The output of the linear regression for Dior shows that the coefficients for the number of new celebrities and traditional celebrities are -0.01011589 and 0.04058056, respectively. This means that for every one unit increase in the number of new celebrities, the natural logarithm of the number of likes received decreases by 0.01011589 units, and for every one unit increase in the number of traditional celebrities, the natural logarithm of the number of likes received increases by 0.04058056 units